app/code/Magento/Inventory/Model/StockRepository/GetList.php

```diff
@@ -0,0 +1,80 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Inventory\Model\StockRepository;
+
+use Magento\Framework\Api\SearchCriteria\CollectionProcessorInterface;
+use Magento\Framework\Api\SearchCriteriaBuilder;
+use Magento\Framework\Api\SearchCriteriaInterface;
+use Magento\Inventory\Model\ResourceModel\Stock\Collection;
+use Magento\Inventory\Model\ResourceModel\Stock\CollectionFactory;
+use Magento\InventoryApi\Api\Data\StockSearchResultsInterface;
+use Magento\InventoryApi\Api\Data\StockSearchResultsInterfaceFactory;
+
+class GetList
```

Comment:
API and SPI segregation
- create spi interfaces for all actions
- implementation
- repository will be proxy to commads